web/rainmaker/dev-packages/egov-billamend-dev/src/ui-config/screens/specs/utils/index.js

```diff
@@ -232,7 +235,88 @@ export const showHideAdhocPopup = (state, dispatch) => {
     handleField("search", "components.adhocDialog", "props.open", !toggle)
   );
 };
+export const getRequiredDocData = async (action, dispatch, moduleDetails, closePopUp) => {
+  let tenantId =
+    process.env.REACT_APP_NAME === "Citizen" ? JSON.parse(getUserInfo()).permanentCity : getTenantId();
+  let mdmsBody = {
+    MdmsCriteria: {
+      tenantId:commonConfig.tenantId,
+      moduleDetails: moduleDetails
+    }
+  };
+  try {
+    let payload = null;
+    payload = await httpRequest(
+      "post",
+      "/egov-mdms-service/v1/_search",
+      "_search",
+      [],
+      mdmsBody
+    );
+    const moduleName = moduleDetails[0].moduleName;
+    let documents = get(
+      payload.MdmsRes,
+      `${moduleName}.documentObj`,
+      []
+    );
 
+    if (moduleName === "PropertyTax") {
+      payload.MdmsRes.tenant.tenants = payload.MdmsRes.tenant.citymodule[1].tenants;
+    }
+    const reqDocuments = getRequiredDocuments(documents, moduleName, footerCallBackForRequiredDataModal(moduleName, closePopUp));
+    set(
+      action,
+      "screenConfig.components.adhocDialog.children.popup",
+      reqDocuments
+    );
+    dispatch(prepareFinalObject("searchScreenMdmsData", payload.MdmsRes));
+    return { payload, reqDocuments };
+  } catch (e) {
+    console.log(e);
+  }
+};
+const footerCallBackForRequiredDataModal = (moduleName, closePopUp) => {
+  const tenant = getTenantId();
+  switch (moduleName) {
+    case "FireNoc":
+      return (state, dispatch) => {
```

Comment:
You can remove this check for module name as it is specific to bill amend and check on click of apply it directs to apply screen